### PR TITLE
Add FIPS indicator for TEST-RAND RNG

### PR DESCRIPTION
**Pull request by paulidale**
This RNG is not and never will be approved and is guarded by a `fips="no"` property but it still should have an indicator saying _not FIPS_.

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
Needs update to EVP_RAND-TEST_RAND.pod
This should also trigger the callback somewhere so you may want to add a call also call ossl_FIPS_IND_callback inside a #ifdef FIPS_MODULE

**Comment by paulidale** [comment]
> This should also trigger the callback somewhere so you may want to add a call to ossl_FIPS_IND_callback inside a #ifdef FIPS_MODULE

Why?  This is never FIPS approved and doesn't fail if it is used.  I don't see a need for the callback.

**Review by t8m — APPROVED** [review]

File: doc/man7/EVP_RAND-TEST-RAND.pod
Comment (slontis) [inline doc/man7/EVP_RAND-TEST-RAND.pod:49]: The comment underneath is not relevant to this getter

File: doc/man7/EVP_RAND-TEST-RAND.pod
Comment (slontis) [inline doc/man7/EVP_RAND-TEST-RAND.pod:25]: hmm am I supposed to be using <integer> not <int>?

**Comment by paulidale** [comment]
Ping @openssl/committers for review.

File: doc/man3/EVP_RAND.pod
Comment (tom-cosgrove-arm) [inline doc/man3/EVP_RAND.pod:270]: I don't see `OSSL_RAND_PARAM_FIPS_APPROVED_INDICATOR ` anywhere in the code, and I don't see `OSSL_DRBG_PARAM_FIPS_APPROVED_INDICATOR` in the `.pod` files - is that intentional?

Comment (paulidale) [reply doc/man3/EVP_RAND.pod:270]: Damn aliases.

These are defined in `util/perl/OpenSSL/paramnames.pm` albeit without the `OSSL_` prefix (which the script adds & I really wish I hadn't done that way and will eventually address).

I'll change things to be more consistent.

Comment (paulidale) [reply doc/man3/EVP_RAND.pod:270]: Done I hope.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by tom-cosgrove-arm** [comment]
@t8m can you reconfirm?

**Review by slontis — APPROVED** [review]

**Comment by slontis** [comment]
looks like there is a doc-nits issue

File: doc/man7/EVP_RAND-TEST-RAND.pod
Comment (tom-cosgrove-arm) [inline doc/man7/EVP_RAND-TEST-RAND.pod:1]: Heh, this blank line needs to be deleted, I think

Comment (paulidale) [reply doc/man7/EVP_RAND-TEST-RAND.pod]: Gone now.  I'm not going to request reconfirmations and timer resets.

**Review by tom-cosgrove-arm — APPROVED** [review]
reconfirming anyway :)

**Comment by slontis** [comment]
reconfirmed after trivial cvhange

**Comment by paulidale** [comment]
Merged, thanks for the input.

**Pull request closed without merging**
